test/integration/registerable_test.rb

```diff
@@ -271,6 +271,37 @@ def user_sign_up
     assert User.to_adapter.find_all.empty?
   end
 
+  test 'a signed in user should be able to cancel their account with current password if password required' do
+    swap Devise, require_password_to_destroy: true  do
+      sign_in_as_user
+      get edit_user_registration_path
+
+      within(".destroy-password") do
+        fill_in 'current password', with: '12345678'
+      end
+
+      click_button "Cancel my account"
+      assert_contain "Bye! Your account has been successfully cancelled. We hope to see you again soon."
+    end
+    assert User.to_adapter.find_all.empty?
+  end
+
+  test 'a signed in user should not be able to cancel their account with incorrect password if password required' do
+    swap Devise, require_password_to_destroy: true  do
+      sign_in_as_user
+      get edit_user_registration_path
+
+      within(".destroy-password") do
+        fill_in 'current password', with: '87654321'
+      end
+
+      click_button "Cancel my account"
+      assert_contain "Current password is invalid"
+
+      assert_current_url '/users'
```

Comment:
Maybe assert that the resource wasn't deleted, the opposite of line 286.

Not sure about the project code style but something like:

```ruby
refute User.to_adapter.find_all.empty?
```

Reply:
here, I would only assert things that the User is seeing since it's seems a feature test and include (if there is no one) a unit test to ensure that user is deleted or not. So, I would suggest to keep it as it is and changing the assert above to follow the same approach.

Reply:
In that case, the spec above would have to try to log in again and see that the account doesn't exist anymore.

Reply:
> here, I would only assert things that the User is seeing since it's seems a feature test and include (if there is no one) a unit test to ensure that the user is deleted or not. So, I would suggest to keep it as it is and changing the assert above to follow the same approach.

I inserted some tests for the destroy action, previously there were none, but I'm having a problem for the success case, working on it to see what can I do more.

Reply:
> In that case, the spec above would have to try to log in again and see that the account doesn't exist anymore.

Definitely a interesting thing to check, I'll improve this. But the `refute` works well too.

Reply:
I'm ok if we use `refute` here to ensure that the user isn't created in this scenario.

I just point this out because, recently, we tried to move from webrat to capybara but we aborted because it requires a lot of changes. One of the reasons is that our test suite is mixing integration tests with features tests and webrat works a little different from capybara in this aspect. But reading the test again, I don't think it would be a problem at all.